pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -184,6 +183,13 @@ func buildClusterHosts(env *model.Environment, service *model.Service, port int)
 
 	hosts := make([]*core.Address, 0)
 	for _, instance := range instances {
+		// Only send endpoints from the networks in the network view requested by the proxy.
+		// The default network view assigned to the Proxy is the UnnamedNetwork (""), which matches
+		// the default network assigned to endpoints that don't have an explicit network
+		if _, found := proxyNetworkView[instance.Endpoint.Network]; !found {
```

Comment:
This is `map[string]bool` and `bool` default value is `false` - the key is not in the map you get the default value, so this can be simplified to:

```go
if !proxyNetworkView[instance.Endpoint.Network] {
	continue
}
```